Media track settings & constraints

regexident opened the pull request:
This adds an implementation of [media track settings](https://w3.org/TR/mediacapture-streams/#media-track-settings), [media track constraints](https://w3.org/TR/mediacapture-streams/#media-track-constraints),  including constraint merging and the [fitness distance algorithm](https://www.w3.org/TR/mediacapture-streams/#dfn-fitness-distance).

Unit tests coming up next. Wanted to get the API figured out first. 🙂

rainliu commented:
hi @regexident, I added you into media team with write access right to media repo. You can directly push code into media repo, as long as not breaking media/io submodule that is required by webrtc v0.1.0

regexident commented:
Added a bunch of unit tests (104 total), builder patterns (where it made sense) and simplified the implementations of the fitness distance algorithm for numeric/non-numeric settings.
